src/cogent3/core/alignment.py

```diff
@@ -914,8 +914,8 @@ def take_seqs(self, seqs, negate=False, **kwargs):
         result = self.__class__(result, names=seqs, info=self.info, **kwargs)
         if self.annotation_db:
             result.annotation_db = type(self.annotation_db)()
-            result.annotation_db.update(
-                annot_db=self.annotation_db, seqids=result.names
+            result.annotation_db._update_db_from_other_db(
+                other_db=self.annotation_db, seqids=result.names
```

Comment:
Never use a private method in another class. Change the implementation of `annotation_db.update()` to using the new method instead of `_update_from_rich_dict()`